### ISSUE TITLE
Code not work on esp8266.

### ISSUE THREAD

**Issue by brightproject** [cbm80amiga/N5110_GUI_encoder_demo#1]
Help with code adaptation with esp8266 under the library Adafruit_PCD8544

**Comment by cbm80amiga** [comment]
Yes it doesn't and it will not work because it uses my 5110 library which doesn't use any RAM for frame buffer and AVR specific deep-sleep code.

**Comment by brightproject** [comment]
Tell me which pieces of code to rewrite for the Adafruit_PCD8544 library?
For example, the battery display is difficult to understand.
```
void drawBattBig(int x, int y, int wd, int perc)
{
  int w = wd*perc/100;
  lcd.fillWin(x,y+0,1+w,1,B11111110);
  lcd.fillWin(x,y+1,1+w,1,B01111111);
  x+=w+1;
  w=wd-w;
  if(w>0) {
    lcd.fillWin(x,y+0,w,1,B00000010);
    lcd.fillWin(x,y+1,w,1,B01000000);
    x+=w;
  }
  lcd.fillWin(x,y+0,1,1,B11111110);
  lcd.fillWin(x,y+1,1,1,B01111111); x++;
  lcd.fillWin(x,y+0,1,1,B11110000);
  lcd.fillWin(x,y+1,1,1,B00001111); x++;
  lcd.fillWin(x,y+0,1,1,B11110000);
  lcd.fillWin(x,y+1,1,1,B00001111);
}
```
I used Adafruit_PCD8544 library, and this is simple.
Are you used old technologies?
**N5110_SPI**

```
void displayIntMenuPage(String menuItem, int value)
{
    display.setTextSize(1);
    display.clearDisplay();
    display.setTextColor(BLACK, WHITE);
    display.setCursor(15, 0);
    display.print(utf8rus(menuItem));
    display.drawFastHLine(0,10,83,BLACK);
    display.setCursor(5, 15);
    display.print(utf8rus("Temperature"));
    display.setTextSize(2);
    display.setCursor(5, 25);
    display.print(value);
    display.setCursor(5, 35);
    display.print(utf8rus("\xB0\C"));
    display.display();
}
```

**Comment by cbm80amiga** [comment]
fillWin(x,line,w,nlines,val) fills directly frame buffer window in N5110 starting from x,line. Look at my N5110 library for more info and examples.

**Comment by brightproject** [comment]
You are using such a complex library for rendering to the display.
Does it increase the speed of working with the display or do you just like complexity)?
Or did you write the code to make it difficult to copy and repeat?
Could you write an example for a couple of menu items with graphs?
Using the libraries for this:
```
#include <Adafruit_GFX.h>
#include <Adafruit_PCD8544.h>
```
I would be extremely grateful to you!

**Comment by brightproject** [comment]
Please tell me the library N5110 library only works with atmega328p?
You use in the library software or hardware SPI?
For the library to work, you need to connect the MOSI and SCK pins to the hardware pins of the microcontroller?
I'm trying to adapt your code for esp8266, but so far nothing has worked.

**Comment by brightproject** [comment]
> fillWin(x,line,w,nlines,val) fills directly frame buffer window in N5110 starting from x,line. Look at my N5110 library for more info and examples.

Pls help converted method on this library
```
int N5110_SPI::fillWin(int x, uint8_t y8, uint8_t wd, uint8_t ht8, uint8_t data)
{
  ALIGNMENT;
  for(int i=0; i<ht8; i++) {
    gotoXY(x,y8+i);
    setDat();
    startCS();
    for(int j=0; j<wd; j++) sendData(data);
    stopCS();
  }
  return x+wd;
}
```
For changes
`lcd.fillWin(0, 2, encoderPos, 1, 0xfc);`
On this library
https://github.com/maxint-rd/I2C-PCF8574-PCD8544-Nokia-5110-LCD/blob/master/PCF8574_PCD8544.cpp
Adafruit nokia 5110
https://github.com/adafruit/Adafruit-PCD8544-Nokia-5110-LCD-library
Which of the methods for displaying information can be used from my library for Esp8266?
```
void testdrawrect(void) {
  for (int16_t i=0; i<display.height()/2; i+=2) {
    display.drawRect(i, i, display.width()-2*i, display.height()-2*i, BLACK);
    display.display();
  }
}
```
or
```
void testdrawchar(void) {
  display.setTextSize(1);
  display.setTextColor(BLACK);
  display.setCursor(0,0);

  for (uint8_t i=0; i < 168; i++) {
    if (i == '\n') continue;
    display.write(i);
    //if ((i > 0) && (i % 14 == 0))
      //display.println();
  }    
  display.display();
}
```

**Comment by cbm80amiga** [comment]
I used my N5110 library in this example. And my library doesn't use RAM for local frame buffer - it writes directly to LCD's internal frame buffer. So other libraries are incompatible.

**Comment by brightproject** [comment]
You can write comments to the function **fillWin**?
`int fillWin(int x, uint8_t y8, uint8_t wd, uint8_t ht8, uint8_t data);`
and
```
int N5110_SPI::fillWin(int x, uint8_t y8, uint8_t wd, uint8_t ht8, uint8_t data)
{
  ALIGNMENT;
  for(int i=0; i<ht8; i++) {
    gotoXY(x,y8+i);
    setDat();
    startCS();
    for(int j=0; j<wd; j++) sendData(data);
    stopCS();
  }
  return x+wd;
}
```
What data is in the variables?
**int x, uint8_t y8, uint8_t wd, uint8_t ht8, uint8_t data**

**fillWin** is short for window filling?

Please show with an example how to make a vertical scroll(slider), which depends on the selected menu item and dynamically changes after moving through the menu.

```
void drawMenuSlider()
{
  int y, n = (8*numScrLines-2-5-2)*menuLine/(numMenus-1);
  scrWd = 3;
  scrHt = numScrLines;
  clrBuf();
  for(y=0; y<numScrLines*8; y++) drawPixel(1,y,1);
  for(y=0; y<5; y++) { drawPixel(0,y+n+2,1); drawPixel(2,y+n+2,1); }
  lcd.drawBuf(scr,81,0,scrWd,scrHt);
}
```

How to display the position of the pixels of this slider using the library Adafruit PCD8544 https://github.com/adafruit/Adafruit-PCD8544-Nokia-5110-LCD-library/blob/master/Adafruit_PCD8544.h

https://www.sparkfun.com/datasheets/LCD/Monochrome/Nokia5110.pdf

Does your library draw per pixel or per character?

https://i.ibb.co/smc3Zgy/18-10-2022-14-10-07.png

https://i.ibb.co/xmzngt9/18-10-2022-14-08-24.png

Please help with code, I really want to apply your ideas to ESP8266.
